src/frontend/components/UI/QuestsViewer/components/QuestDetailsWrapper/index.tsx

```diff
@@ -141,20 +206,45 @@ export function QuestDetailsWrapper({
   }
 
   const mintOnChainReward = async (reward: Reward) => {
+    setWarningMessage(undefined)
+
     if (questMeta?.id === undefined) {
-      console.error('tried to mint but quest meta id is undefined')
-      return
+      throw Error('tried to mint but quest meta id is undefined')
     }
+
+    if (reward.chain_id === null) {
+      throw Error('chain id is not set for reward when trying to mint')
+    }
+
     if (account.address === undefined) {
-      console.error('tried to mint but no account connected')
+      setWarningMessage('Please connect your wallet to claim rewards.')
+      return
+    }
+
+    if (!walletBalance) {
+      throw Error('Wallet balance not available')
+    }
+
+    await switchChainAsync({ chainId: reward.chain_id })
+
+    const gasNeeded = await getRewardClaimGasEstimation(reward)
+    const hasEnoughBalance = walletBalance.value >= gasNeeded
+
+    if (!hasEnoughBalance) {
+      window.api.logError(
+        `Not enough balance in the connected wallet to cover the gas fee associated with this Quest Reward claim. Current balance: ${walletBalance.value}, gas needed: ${gasNeeded}`
+      )
+      setWarningMessage(
+        'Not enough balance in the connected wallet to cover the gas fee associated with this Quest Reward claim.'
```

Comment:
should we translate this string? not sure if gas fee will translate correctly

Reply:
good call

Reply:
Would be good to check with metamask, no? Maybe this is translated for some languages, I imagine for instance languages like Chinese, Korean or Russian or any other that don't use Latin chars @BrettCleary @eliobricenov